webapp/src/pages/boardPage/boardPage.tsx

```diff
@@ -102,10 +107,18 @@ const BoardPage = (props: Props): JSX.Element => {
 
     // TODO: Make this less brittle. This only works because this is the root render function
     useEffect(() => {
-        UserSettings.lastTeamId = teamId
-        octoClient.teamId = teamId
-        dispatch(setTeam(teamId))
-    }, [teamId])
+        if (allTeams.length === 0) {
+            return
+        }
+        const isValidTeam = allTeams.some((team) => team.id === match.params.teamId) || match.params.teamId === Constants.globalTeamId
+        if (isValidTeam) {
+            UserSettings.lastTeamId = teamId
+            octoClient.teamId = teamId
+            dispatch(setTeam(teamId))
+        } else {
+            dispatch(setGlobalError('team-undefined'))
```

Comment:
I'm not sure what this does, but for example if I open the url `http://localhost:8065/boards/team/invalid-team-id` on this branch, there are no errors and an empty template selector screen-

<img width="2555" height="2160" alt="Image" src="https://github.com/user-attachments/assets/1c0dbd90-e21f-43d3-87cf-af74b94ddd51" />

Whereas if I try to do the same thing on Hub, I get the screen with right error message-

<img width="1590" height="1274" alt="Image" src="https://github.com/user-attachments/assets/7b72c6e5-724d-498e-bfce-0b8bc8fda321" />